types/src/system_contract_errors/auction.rs

```diff
@@ -189,3 +193,25 @@ impl From<PurseLookupError> for Error {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::convert::TryFrom;
+
+    use super::{Error, MAX_ERROR_VALUE};
+
+    #[test]
+    fn error_round_trips() {
+        for i in 0..=u8::max_value() {
+            match Error::try_from(i) {
+                Ok(error) if i <= MAX_ERROR_VALUE => assert_eq!(error as u8, i),
+                Ok(error) => panic!(
+                    "value of variant {} ({}) exceeds MAX_ERROR_VALUE ({})",
+                    error, i, MAX_ERROR_VALUE
+                ),
+                Err(_) if i > MAX_ERROR_VALUE => (),
```

Comment:
Maybe `Err(TryFromU8ForError(())) => assert!(i > MAX_ERROR_VALUE)` here?

Reply:
Yes I suppose that would force attention here if someone were to change the `Error` type in our `TryFrom` instance, which is probably a good thing.  Will do on both `Err` variants.

Reply:
Fixed.